server-common/src/main/java/org/apache/cassandra/sidecar/common/server/utils/DurationSpec.java

```diff
@@ -0,0 +1,322 @@
+        {
+            this.quantity = Long.parseLong(matcher.group(1));
+            this.unit = fromSymbol(matcher.group(2));
+        }
+        else
+        {
+            throw iae(value);
+        }
+
+        validateMinUnit(value, unit, minimumUnit());
+        validateQuantity(value, quantity, unit, minimumUnit());
+    }
+
+    /**
+     * Constructs the {@link DurationSpec} with the given {@code quantity} and {@code unit}.
+     *
+     * @param quantity the quantity for the duration
+     * @param unit     the unit for the duration
+     * @throws IllegalArgumentException when the unit is invalid or the quantity is invalid
+     */
+    protected DurationSpec(long quantity, TimeUnit unit) throws IllegalArgumentException
+    {
+        this.quantity = quantity;
+        this.unit = unit;
+
+        validateMinUnit(this, unit, minimumUnit());
+        validateQuantity(this, quantity, unit, minimumUnit());
+    }
+
+    /**
+     * @return the minimum unit that can be represented by this type
+     */
+    public abstract TimeUnit minimumUnit();
+
+    /**
+     * @return the amount of time in the specified unit
+     */
+    public long quantity()
+    {
+        return quantity;
+    }
+
+    /**
+     * @return the time unit to specify the amount of duration
+     */
+    public TimeUnit unit()
+    {
+        return unit;
+    }
+
+    /**
+     * Converts this duration spec to the {@code targetUnit}.
+     * Conversions from finer to coarser granularities lose precision.
+     *
+     * <p>Conversions from coarser to finer granularities with arguments
+     * that would numerically overflow saturate to {@link Long#MIN_VALUE}
+     * if negative or {@link Long#MAX_VALUE} if positive.
+     *
+     * @param targetUnit the target conversion unit
+     * @return the converted duration in the {@code targetUnit},
+     * or {@code Long.MIN_VALUE} if conversion would negatively overflow,
+     * or {@code Long.MAX_VALUE} if it would positively overflow.
+     */
+    public long to(TimeUnit targetUnit)
+    {
+        return targetUnit.convert(quantity(), unit());
+    }
+
+    /**
+     * @return the duration in seconds
+     */
+    public long toSeconds()
+    {
+        return to(TimeUnit.SECONDS);
+    }
+
+    /**
+     * @return the duration in milliseconds
+     */
+    public long toMillis()
+    {
+        return to(TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * @return the duration in milliseconds returned as an integer, if the value overflows,
+     * returns {@link Integer#MAX_VALUE}
+     */
+    public int toIntMillis()
+    {
+        long longMillis = toMillis();
+        if (longMillis > Integer.MAX_VALUE)
+        {
+            return Integer.MAX_VALUE;
+        }
+        return (int) longMillis;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode()
+    {
+        // Milliseconds seems to be a reasonable tradeoff
+        return Objects.hash(unit.toMillis(quantity));
```

Comment:
NIT: Can now be replaced with a call to `toMillis`
```suggestion
        return Objects.hash(toMillis());
```